packages/flutter/lib/src/material/ink_well.dart

```diff
@@ -64,6 +64,54 @@ abstract class InteractiveInkFeature extends InkFeature {
     _color = value;
     controller.markNeedsPaint();
   }
+
+  /// Called during [paintFeature] in [InkSplash] and [InkRipple] widgets
+  ///
```

Comment:
In between, as @HansMuller has recommended, there should be some documentation here on what the named parameters do since this function will be public. Assume that readers do not have any context to what this function is used for. Here's an example to get started:
```dart
/// If [customBorder] not null, the canvas is clipped by the [ShapeBorder] provided. If 
/// [customBorder] is null and [borderRadius] is non-null and not [BorderRadius.zero],
/// the canvas is clipped by the [BorderRadius] provided. {and so forth}
```

Reply:
The rest of the parameters are still missing explanations/comments (canvas, transform, paint, center, etc).

Reply:
@shihaohong Added all remaining explanations except `canvas` as that seems redundant in case of painting